csfieldguide/static/interactives/bias-in-big-data/js/bias-in-big-data.js

```diff
@@ -0,0 +1,159 @@
+        // add glow around clicked circle
+        $(this).addClass('glow');
+    });
+    $('#next-stage').removeClass('d-none');
+    $('#start-again').addClass('d-none');
+    createSlider();
+    bgColourSlider[0].noUiSlider.on('update', updateSlider);
+}
+
+
+function createSlider() {
+    noUiSlider.create(bgColourSlider[0], {
+        start: sliderStartPos,
+        step: 1,
+        connect: "lower",
+        orientation: "horizontal",
+        range: {
+          'min': SLIDER_MIN,
+          'max': SLIDER_MAX
+        },
+        format: wNumb({
+          decimals: 0
+        })
+    });
+}
+
+
+function updateSlider() {
+    var value = bgColourSlider[0].noUiSlider.get();
+    hslColour = 'hsl(' + value + ', 100%, 50%)';
+    $('#circles-area').css('background', hslColour);
+    $('.noUi-handle').css('background', hslColour);
+}
+
+
+function getRandomPosition() {
+    var circlesAreaHeight = $('#circles-area').height();
+    var circlesAreaWidth = $('#circles-area').width();
+    var randHeight = Math.floor((Math.random() * circlesAreaHeight));
+    var randWidth = Math.floor((Math.random() * circlesAreaWidth));
+
+    // convert px to %
+    heightInPercentage = Math.floor((randHeight / circlesAreaHeight) * 100);
+    widthInPercentage = Math.floor((randWidth / circlesAreaWidth) * 100);
+
+    // reduces or increases percentage by 5% to prevent circles going outside of parent
+    if (heightInPercentage >= PERENTAGE_BOUNDARY_UPPER) {
+        heightInPercentage -= PERCENTAGE_ADJUSTMENT;
+    } else if (heightInPercentage <= PERENTAGE_BOUNDARY_LOWER) {
+        heightInPercentage += PERCENTAGE_ADJUSTMENT
+    }
+    if (widthInPercentage >= PERENTAGE_BOUNDARY_UPPER) {
+        widthInPercentage -= PERCENTAGE_ADJUSTMENT;
+    } else if (widthInPercentage <= PERENTAGE_BOUNDARY_LOWER) {
+        widthInPercentage += PERCENTAGE_ADJUSTMENT;
+    }
+
+    return {
+        top: heightInPercentage + '%',
+        left: widthInPercentage + '%'
+    };
+}
+
+
+function createCircle(colour) {
+    var colour =  colour || getRandomColour();
+    var $circle = $("<div>").addClass('circle dynamic d-none ' + colour);
+    $circle.css(getRandomPosition());
+    $('#circles-area').append($circle);
+}
+
+
+function loadNextStage() {
+    if (firstStage) {
+        // what happens if they haven't missed any circles?
+        $('#circles-area').removeClass(startColour);
+        $('#circles-area').addClass('grey');
+        $('#instruction-text').html(MISSED_CIRCLES_TEXT);
+        firstStage = false;
+    } else {
+        $('#instruction-text').html(SLIDER_TEXT);
+        $('.circle').removeClass('glow');
+        $('#circles-area').removeClass('grey');
+        $('#circles-area').addClass(startColour);
+        $('#next-stage').addClass('d-none');
+        $('#start-again').removeClass('d-none');
+        $('#background-colour-slider-container').removeClass('d-none');
+        // disable click event for circles
+        $('.circle').off('click');
```

Comment:
Is there reason to disable the clicking? They aren't being counted anyway so I don't see why

Reply:
Mostly so that when they change the background colour with the slider they can visibly see the circles disappearing and reappearing. The glow would give that away.

Reply:
The glow giving that away is not necessarily a bad thing though. For example a teacher could do the slidering and show dots disappear, then click dots and slider again to show they're still there - The user doesn't have to click on them

Reply:
Or it could be a toggle?